link/src/main/java/com/stripe/android/link/ui/verification/VerificationScreen.kt

```diff
@@ -1,12 +1,228 @@
+import com.stripe.android.link.R
+import com.stripe.android.link.theme.linkColors
+import com.stripe.android.link.theme.linkShapes
+import com.stripe.android.link.ui.ErrorText
+import com.stripe.android.link.ui.ScrollableTopLevelColumn
+import com.stripe.android.uicore.DefaultStripeTheme
+import com.stripe.android.uicore.elements.OTPElement
+import com.stripe.android.uicore.elements.OTPElementUI
+import com.stripe.android.uicore.utils.collectAsState
 
-@SuppressWarnings("UnusedParameter")
 @Composable
 internal fun VerificationScreen(
     viewModel: VerificationViewModel
 ) {
-    Text(text = "VerificationScreen")
+    val state by viewModel.viewState.collectAsState()
+    val context = LocalContext.current
+    val focusManager = LocalFocusManager.current
+    val focusRequester: FocusRequester = remember { FocusRequester() }
+    val keyboardController = LocalSoftwareKeyboardController.current
+
+    LaunchedEffect(state.isProcessing) {
+        if (state.isProcessing) {
+            focusManager.clearFocus(true)
+            keyboardController?.hide()
+        }
+    }
+
+    LaunchedEffect(state.requestFocus) {
+        if (state.requestFocus) {
+            focusRequester.requestFocus()
+            keyboardController?.show()
+            viewModel.onFocusRequested()
+        }
+    }
+
+    LaunchedEffect(state.didSendNewCode) {
+        if (state.didSendNewCode) {
+            Toast.makeText(context, R.string.stripe_verification_code_sent, Toast.LENGTH_SHORT).show()
+            viewModel.didShowCodeSentNotification()
+        }
+    }
+
+    VerificationBody(
+        state = state,
+        otpElement = viewModel.otpElement,
+        focusRequester = focusRequester,
+        onBack = viewModel::onBack,
+        onChangeEmailClick = viewModel::onChangeEmailClicked,
+        onResendCodeClick = viewModel::resendCode
+    )
+}
+
+@Composable
+internal fun VerificationBody(
+    state: VerificationViewState,
+    otpElement: OTPElement,
+    onBack: () -> Unit,
+    onChangeEmailClick: () -> Unit,
+    onResendCodeClick: () -> Unit,
+    focusRequester: FocusRequester = remember { FocusRequester() },
+) {
+    BackHandler(onBack = onBack)
+
+    ScrollableTopLevelColumn {
+        Text(
+            text = stringResource(R.string.stripe_verification_header),
+            modifier = Modifier
+                .padding(vertical = 4.dp),
+            textAlign = TextAlign.Center,
+            style = MaterialTheme.typography.h2,
+            color = MaterialTheme.colors.onPrimary
+        )
+        Text(
+            text = stringResource(R.string.stripe_verification_message, state.redactedPhoneNumber),
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(top = 4.dp, bottom = 20.dp),
+            textAlign = TextAlign.Center,
+            style = MaterialTheme.typography.body1,
+            color = MaterialTheme.colors.onSecondary
+        )
+        DefaultStripeTheme {
```

Comment:
out of curiosity - why does this theme only apply to the otp element and not to the whole screen? Or why are we not using the link theme for this part of the UI?

Reply:
I left it because the old UI wrapped the component in this theme, but it looks like removing it doesn't change anything. I just took it out.